preliminary equations (#428)
PowerGridModel/power-grid-model: docs/user_manual/components.md

(i.e. `fault_type` is not `FaultType.three_phase`).
```

#### Electric Model

`line` is described by a π model, where 

$$
   \begin{eqnarray}

>       Z_\mathrm{series}    & = & r + \mathrm{j}x \\
>       Y_\mathrm{shunt} & = & 2 \pi fc/(tan \sigma +\mathrm{j})

Reviewer: ```suggestion
      \bar{Z}_\mathrm{series}    & = & r + \mathrm{j}x \\
      \bar{Y}_\mathrm{shunt} & = & 2 \pi fc/(tan \sigma +\mathrm{j})
```

(equivalent to 10e6 siemens for 10kV network). Therefore, it is chosen by design that no sensors can be coupled to a `link`.
There is no additional attribute for `link`.

#### Electric Model: 
`link` is modeled by a constant reactance $Y_\mathrm{series}$, where

$$
   \begin{eqnarray}

>       Y_\mathrm{series}    & = & 1 & \times 10^6&.

Reviewer: ```suggestion
      Y_\mathrm{series}    & = & 1 & \times 10^6 p.u. &.
```

increased.
```

#### Electric Model
`transformer` is described by π model, where z_series and y_shunt can be computed by following equations:

$$

$$

z_base = `sn`/(`u2`^2)

|z_series| = |`uk`|/z_base

Re(z_series) = (`pk`/`sn`)/z_base

> Im(z_series) = sign(`uk`) * sqrt(|z_series|^2-Re(z_series)^2)

Reviewer: From a user perspective, uk is always positive as the transformer must have a positive impedance. (Check the restriction at the components table)

`uk` can be negative theoretically at times for eg, the three winding transformer. The way we model a three winding transformer is by star connected 3 two winding transformers. Then `uk` isconverted from delta to star which might lead to negative value.

base_i_to = base_power_3p / u2_rated / sqrt(3),

base_y_to = base_i_to^2 / base_power_1p

y_series = (1/z_seires) / base_y_to

y_shunt_abs = `i0` * `sn` / `u2`^2

Re(y_shunt) = `p0` / (`u2`^2)

Im(y_shunt) = 0 (if Re(y_shunt) > y_shunt_abs) or = -sqrt(y_shunt_abs^2-Re(y_shunt)^2)

y_shunt = y_shunt / base_y_to

> where base_power_3p = 1e6, and base_power_1p = base_power_3p / 3. u2_rated is a constant value, which is normally set to 10e3.

Reviewer: u2_rated is the `u2` attribute of the transformer and not a constant.

increased.
```

#### Electric Model
`transformer` is described by a π model, where $Z_\mathrm{series}$ can be computed as

$$
    \begin{eqnarray} 
        & |Z_\mathrm{series}| = uk / z_\mathrm{base}\\ 
        &\mathrm{Re}(Z_\mathrm{series}) = (pk/sn) / z_\mathrm{base}\\

>         &\mathrm{Im}(Z_\mathrm{series}) = \sqrt{|Z_\mathrm{series}|^2-\mathrm{Re}(Z_\mathrm{series})^2} p.u. \\

Reviewer: ```suggestion
        &\mathrm{Im}(Z_\mathrm{series}) = \sqrt{|Z_\mathrm{series}|^2-\mathrm{Re}(Z_\mathrm{series})^2} \\
```

    \end{eqnarray}
$$

and $Y_\mathrm{shunt}$ can be computed as

$$
    \begin{eqnarray} 
        &|Y_\mathrm{shunt}| = i0 / y_\mathrm{base} \\
        &\mathrm{Re}(Y_\mathrm{shunt}) = (sn/p0) / y_\mathrm{base} \\
        &\mathrm{Im}(Y_\mathrm{shunt}) = 
        \begin{cases}
            0 & \text{ if $\mathrm{Re}(Y_\mathrm{shunt})$ > $|Y_\mathrm{shunt}|$} \\

>             -\sqrt{|Y_\mathrm{shunt}|^2-\mathrm{Re}(Y_\mathrm{shunt})^2} p.u. & \text{otherwise} 

Reviewer: ```suggestion
            -\sqrt{|Y_\mathrm{shunt}|^2-\mathrm{Re}(Y_\mathrm{shunt})^2} & \text{otherwise} 
```

increased.
```

#### Electric Model
`transformer` is described by a π model, where $z_{series}$ and $y_{shunt}$ can be computed by following equations:

$$
   \begin{eqnarray} 

>         z_{base} = sn/(u2^2),

Reviewer: It would be nice to highlight properties actually in use. 
(Also remove ones which are not wherever possible) The details about base are for our understanding and user wont have to concern with. For eg. `base_power_3p` can definitely be removed, while other bases can be kept if they make understanding better.

increased.
```

#### Electric Model
`transformer` is described by a π model, where $Z_{series}$ and $Y_{shunt}$ can be computed by following equations:

$$

>    \begin{eqnarray} 
>         z_{base} = sn/(u2^2)\\
>         |Z_{series}| = uk/z_{base}\\ 
>         Re(Z_{series}) = (pk/sn)/z_{base},\\
>         Im(Z_{series}) = uk * \sqrt{|Z_{series}|^2-Re(Z_{series})^2},\\
>         base\_ i\_ to = base\_ power\_ 3p / u2 / sqrt(3),\\
>         base\_ y\_ to = base\_ i\_ to^2 / base\_ power\_ 1p,\\
>         Y_{series} = (1/Z_{seires}) / base\_ y\_ to,\\
>         |Y_{shunt}| = i0 * sn / u2^2\\
>         Re(Y_{shunt}) = p0 / (u2^2)\\
>         Im(Y_{shunt}) = 0 (if Re(Y_{shunt}) > |Y_{shunt}|) or = -sqrt(|Y_{shunt}|^2-Re(Y_{shunt})^2)\\
>         Y_{shunt} = Y_{shunt} / base\_ y\_ to\\
>    \end{eqnarray}

Reviewer: ```suggestion
   \begin{eqnarray} 
        |Z_{series}| = uk p.u.\\ 
        Re(Z_{series}) = (pk/sn),\\
        Im(Z_{series}) = sqrt{|Z_{series}|^2-Re(Z_{series})^2},\\
        base\_ i\_ to = base\_ power\_ 3p / u2 / sqrt(3),\\
        base\_ y\_ to = base\_ i\_ to^2 / base\_ power\_ 1p,\\
        |Y_{shunt}| = i0 p.u. \\
        Re(Y_{shunt}) = sn/p0 p.u. \\
        Im(Y_{shunt}) = 0 (if Re(Y_{shunt}) > |Y_{shunt}|) or = -sqrt(|Y_{shunt}|^2-Re(Y_{shunt})^2) p.u. \\
   \end{eqnarray}
```

Confirm Re(yshunt) once

> $$
>    \begin{eqnarray} 
>         z_{base} = sn/(u2^2)\\
>         |Z_{series}| = uk/z_{base}\\ 
>         Re(Z_{series}) = (pk/sn)/z_{base},\\
>         Im(Z_{series}) = uk * \sqrt{|Z_{series}|^2-Re(Z_{series})^2},\\
>         base\_ i\_ to = base\_ power\_ 3p / u2 / sqrt(3),\\
>         base\_ y\_ to = base\_ i\_ to^2 / base\_ power\_ 1p,\\
>         Y_{series} = (1/Z_{seires}) / base\_ y\_ to,\\
>         |Y_{shunt}| = i0 * sn / u2^2\\
>         Re(Y_{shunt}) = p0 / (u2^2)\\
>         Im(Y_{shunt}) = 0 (if Re(Y_{shunt}) > |Y_{shunt}|) or = -sqrt(|Y_{shunt}|^2-Re(Y_{shunt})^2)\\
>         Y_{shunt} = Y_{shunt} / base\_ y\_ to\\
>    \end{eqnarray}
> $$
> 
> where base_power_3p = $1 \times 10^6$, and base_power_1p = base_power_3p / 3. 

Reviewer: ```suggestion
$$
   \begin{eqnarray} 
        |Z_{series}| = uk p.u.\\ 
        Re(Z_{series}) = (pk/sn),\\
        Im(Z_{series}) = sqrt{|Z_{series}|^2-Re(Z_{series})^2},\\
        |Y_{shunt}| = i0 p.u. \\
        Re(Y_{shunt}) = sn/p0 p.u. \\
        Im(Y_{shunt}) = 0 (if Re(Y_{shunt}) > |Y_{shunt}|) or = -sqrt(|Y_{shunt}|^2-Re(Y_{shunt})^2) p.u. \\
   \end{eqnarray}
$$

where base_power_3p = $1 \times 10^6$, and base_power_1p = base_power_3p / 3. 
```

Need to confirm Re yshunt once

Reviewer: Also add that the scaling of these parameters happen in a linear fashion for different tap positions

Reviewer: Or better:

```suggestion
$$
   \begin{eqnarray} 
        |Z_{series}| = uk / z_{base}.\\ 
        Re(Z_{series}) = (pk/sn) / z_{base} ,\\
        Im(Z_{series}) = sqrt{|Z_{series}|^2-Re(Z_{series})^2} p.u. ,\\
        |Y_{shunt}| = i0 / y_{base} \\
        Re(Y_{shunt}) = (sn/p0) / y_{base} \\
        Im(Y_{shunt}) = 0 (if Re(Y_{shunt}) > |Y_{shunt}|) or = -sqrt(|Y_{shunt}|^2-Re(Y_{shunt})^2) \\
        z_{base} = sn/(u2^2)\\
        y_{base} = (u2^2)/sn\\
   \end{eqnarray}
$$

```

| `rx_ratio`    | `double`  | -                | R to X ratio                                       |     &#10060; default 0.1     | &#10060; |    `>= 0`    |
| `z01_ratio`   | `double`  | -                | zero sequence to positive sequence impedance ratio |     &#10060; default 1.0     | &#10060; |    `> 0`     |

#### Electric Model
`source` is modeled by using an internal constant impedance r+jx with positive sequence and zero sequence.

The value of impedance can be computed using following equations:

For positive sequence,

> z_s = s_base/sk

Reviewer: Got left out: mention z_s as source impedance or make z_source.

| `p_specified` | `RealValueInput` | watt (W)                   | specified active power   | &#10024; only for power flow | &#10004; |
| `q_specified` | `RealValueInput` | volt-ampere-reactive (var) | specified reactive power | &#10024; only for power flow | &#10004; |

##### Electric model

`generic_load_gen` are modelled by using the so-called ZIP load model in power-grid-model, 
where a load/generator is represented as a composition of constant power (P), constant current (I) and constant impedance (Z).

The injection current I_inj of each ZIP model type can be computed from s = `p_specified` + j`q_specified`as:

> For constant impedance (Z) load: I_inj = S*u

Reviewer: mention that u here is calculated node voltage and it is complex of u and u_angle attribute of node

| `p_specified` | `RealValueInput` | watt (W)                   | specified active power   | &#10024; only for power flow | &#10004; |
| `q_specified` | `RealValueInput` | volt-ampere-reactive (var) | specified reactive power | &#10024; only for power flow | &#10004; |

##### Electric model

`generic_load_gen` are modelled by using the so-called ZIP load model in power-grid-model, 
where a load/generator is represented as a composition of constant power (P), constant current (I) and constant impedance (Z).

The injection of each ZIP model type can be computed as follows:

- for a constant impedance (Z) load/generator,

> $$
>    \begin{eqnarray} 
>         S = S_\mathrm{specified} \times u^2
>    \end{eqnarray}
> $$
> 
> - for a constant current (I) load/generator,
> 
> $$
>    \begin{eqnarray} 
>         S = S_\mathrm{specified} \times u
>    \end{eqnarray}
> $$
> 
> - for a constant power (P) load/generator:,
> 
> $$
>    \begin{eqnarray} 
>         S = S_\mathrm{specified}
>    \end{eqnarray}
> $$
> 
> where $u$ is the calculated node voltage.

Reviewer: ```suggestion
$$
   \begin{eqnarray} 
        S = S_\mathrm{specified} \times \bar{u}^2
   \end{eqnarray}
$$

- for a constant current (I) load/generator,

$$
   \begin{eqnarray} 
        S = S_\mathrm{specified} \times \bar{u}
   \end{eqnarray}
$$

- for a constant power (P) load/generator:,

$$
   \begin{eqnarray} 
        S = S_\mathrm{specified}
   \end{eqnarray}
$$

where $\bar{u}$ is the calculated node voltage.
```

| `p_specified` | `RealValueInput` | watt (W)                   | specified active power   | &#10024; only for power flow | &#10004; |
| `q_specified` | `RealValueInput` | volt-ampere-reactive (var) | specified reactive power | &#10024; only for power flow | &#10004; |

##### Electric model

`generic_load_gen` are modelled by using the so-called ZIP load model in power-grid-model, 
where a load/generator is represented as a composition of constant power (P), constant current (I) and constant impedance (Z).

> The injection current $I_{inj}$ of each ZIP model type can be computed from s = `p_specified` + j`q_specified`as:
> 
> - for constant impedance (Z) load: 
> 
> $$
>    \begin{eqnarray} 
>         I_{inj} = S * u
>    \end{eqnarray}
> $$
> 
> - for Constant current (I) load: 
> 
> $$
>    \begin{eqnarray} 
>         I_{inj} = S*|u|/u = S
>    \end{eqnarray}
> $$
> 
> - for constant power (P) load: 
> 
> $$
>    \begin{eqnarray} 
>         I_{inj} = S / u
>    \end{eqnarray}
> $$
> 
> where S = `p_specified` + j`q_specified`, and u is calculated node voltage, more specifically a complex representation of `u` and `u_angle` attribute of node.

Reviewer: ```suggestion

- for constant impedance (Z) load: 

$$
   \begin{eqnarray} 
        S = S_{specified} * (u^2)
   \end{eqnarray}
$$

- for Constant current (I) load: 

$$
   \begin{eqnarray} 
        S = S_{specified} * u
   \end{eqnarray}
$$

- for constant power (P) load: 

$$
   \begin{eqnarray} 
        S = S_{specfied}
   \end{eqnarray}
$$

where S_{specified} = `p_specified` + j`q_specified`, S = `p` + j`q` which are the result attributes and u is calculated node voltage, more specifically a complex representation of `u` and `u_angle` attribute of node.
```

| `u_residual`       | `RealValueOutput` | volt (V) | residual value between measured voltage magnitude and calculated voltage magnitude                                       |
| `u_angle_residual` | `RealValueOutput` | rad      | residual value between measured voltage angle and calculated voltage angle (only possible with phasor measurement units) |

#### Electric Model
`generic_voltage_sensor` is modeled by following equations:

u_residual = u_measured - “u_node” ,

u_angle_residual = u_angle_measured - "u_angle_node",

> where “u_node” and "u_angle_node" are the steady state output variables`u` and `u_angle` of corresponding `node`

Reviewer: ```suggestion
where “u_node” and "u_angle_node" are the steady state output variables`u` and `u_angle` of corresponding `node` provided at `measured_object`
```

| `u_residual`       | `RealValueOutput` | volt (V) | residual value between measured voltage magnitude and calculated voltage magnitude                                       |
| `u_angle_residual` | `RealValueOutput` | rad      | residual value between measured voltage angle and calculated voltage angle (only possible with phasor measurement units) |

#### Electric Model
`generic_voltage_sensor` is modeled by following equations:

$$
   \begin{eqnarray} 

>         && u\_ residual = u\_ measured - u\_ node \\
>         && u\_ angle\_ residual = u\_ angle\_ measured - u\_ angle\_ node

Reviewer: ```suggestion
        && u\_ residual = u\_ measured - u\_ state \\
        && u\_ angle\_ residual = u\_ angle\_ measured - u\_ angle\_ state
```

| `p_residual` | `RealValueOutput` | watt (W)                   | residual value between measured active power and calculated active power     |
| `q_residual` | `RealValueOutput` | volt-ampere-reactive (var) | residual value between measured reactive power and calculated reactive power |

#### Electric Model
`Generic Power Sensor` is modeled by following equations:

p_residual = p_measured - “p_node” ,

q_residual = q_measured - “q_node” ,

> where “p_node” and "q_node" are the steady state output variables `p` and `q` of corresponding `node`

Reviewer: ```suggestion
where “p_node” and "q_node" are the steady state output variables `p` and `q` of corresponding to the element at `measured_object`. The type of flow of this power is given by `measured_terminal_type`
```

| `p_residual` | `RealValueOutput` | watt (W)                   | residual value between measured active power and calculated active power     |
| `q_residual` | `RealValueOutput` | volt-ampere-reactive (var) | residual value between measured reactive power and calculated reactive power |

#### Electric Model
`Generic Power Sensor` is modeled by following equations:

$$
   \begin{eqnarray} 

>         p\_ residual = p\_ measured - p\_ node \\
>         q\_ residual = q\_ measured - q\_ node

Reviewer: ```suggestion
        p\_ residual = p\_ measured - p\_ state \\
        q\_ residual = q\_ measured - q\_ state
```

| ----------- | ----------------- | ---------- | ------------- |
| `i_f`       | `RealValueOutput` | ampere (A) | current       |
| `i_f_angle` | `RealValueOutput` | rad        | current angle |

#### Electric Model
Four types of short circuit fault are included in power-grid-model.

| `fault_type`                       | `fault_phase`    | description                                                       |
| ---------------------------------- | ---------------- |-------------------------------------------------------------------|

> | `FaultType.three_phase`            | `FaultPhase.abc` | Three phases are connected with impedance `z` =`r_f` + j`r_f`.    |

Reviewer: Move `z` =`r_f` + j`r_f` above table and everything fits well :)
